pynicotine/transfers.py

```diff
@@ -123,7 +123,7 @@ def __init__(self):
             ("file-download-progress", self._file_download_progress),
             ("file-upload-init", self._file_upload_init),
             ("file-upload-progress", self._file_upload_progress),
-            ("folder-contents-response", self._folder_contents_response),
+            #("folder-contents-response", self._folder_contents_response), Avoid Event as use Download Folder Instead.
```

Comment:
I would keep this code in transfers.py, but modify the `self.requested_folders` dict to store both the destination path and received files (it currently only stores a destination path). 

You can then add the files from the FolderContentsResponse message to the dict (reject them if the folder wasn't requested), and access it with `core.transfers.requested_folders[user][virtual_path]["files"]` in the GUI.

Reply:
Please see other comment. I think removing folder download code from transfer might be easier.